packages/dht/src/dht/contact/SortedContactList.ts

```diff
@@ -102,11 +102,24 @@ export class SortedContactList<C extends { getNodeId: () => DhtAddress }> extend
         }
     }
 
+    /*
+     * Closest first then others in ascending distance order
+     */
     public getClosestContacts(limit?: number): C[] {
         const ret = this.getAllContacts()
         return (limit === undefined) 
             ? ret 
-            : ret.slice(0, limit)
+            : ret.slice(0, Math.max(limit, 0))
+    }
+
+    /*
+     * Furthest first then others in descending distance order
+     */
+    getFurthestContacts(limit?: number): C[] {
+        const ret = this.getClosestContacts().reverse()
```

Comment:
Not urgently important here but:

`reverse()` mutates the original array and returns the mutated array. `toReversed()` returns the reversed array without mutating the original array.

Reply:
Fixed